src/integration/blockchain/lightning/lightning-client.ts

```diff
@@ -34,11 +34,31 @@ export class LightningClient {
 
   constructor(private readonly http: HttpService) {}
 
+  // --- DFX API --- //
+  async getSignMessage(address: string): Promise<string> {
+    return this.http
+      .get<{ message: string }>(`${Config.dfxApiUrl}/auth/signMessage`, {
+        params: { address: address },
+      })
+      .then((m) => m.message)
+      .catch((e) => this.throwHttpException(e));
+  }
+
   // --- LND --- //
   async getLndInfo(): Promise<LndInfoDto> {
     return this.http.get<LndInfoDto>(`${Config.blockchain.lightning.lnd.apiUrl}/getinfo`, this.httpLndConfig());
   }
 
+  async signMessage(message: string): Promise<LndSignatureDto> {
+    return this.http
+      .post<LndSignatureDto>(
+        `${Config.blockchain.lightning.lnd.apiUrl}/signmessage`,
+        { msg: Buffer.from(message, 'ascii').toString('base64') },
+        this.httpLndConfig(),
+      )
+      .catch((e) => this.throwHttpException(e));

```

Comment:
I would only catch and rethrow the exceptions for forwarding calls (LNURL and LNDHub), not for internal calls.